src/pages/availability-panel.tsx

```diff
@@ -18,8 +18,9 @@ const AvailabilityPanel: FC = () => {
     const fetchTasks = async () => {
       try {
         const url = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
-        const response = await fetch({ url });
-        const { tasks } = response.data;
+        const { requestPromise } = fetch({ url });
+        const fetchPromise = await requestPromise;
+        const { tasks } = fetchPromise.data;
```

Comment:
Any specific reason to not use the `useFetch` hook? It is anyway having fetching + cancelling functionality, right?

Reply:
Yeah, `useFetch` can be used.
But that change is not related to this issue or pr, let's create a new refactoring issue for this.